### PR TITLE
For #9552: Fix systemWindowInsetTop deprecation in SDK 30.

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by mcarare** [comment]
Decided to also add extension functions, as I wanted to add them on the Fenix-related issue and thought it would be better to have it here. It's not used anywhere else so I added it in the same package.

File: components/support/utils/src/main/java/mozilla/components/support/utils/ext/WindowInsets.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        @Suppress("DEPRECATION")
+        this.systemWindowInsetLeft
+    }
+
+/**
+ * Returns the bottom system window inset in pixels.
+ */
+fun WindowInsets.bottom(): Int =
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+        this.getInsets(WindowInsets.Type.systemBars()).bottom
```
Comment (Amejia481) [inline components/support/utils/src/main/java/mozilla/components/support/utils/ext/WindowInsets.kt:48]: Nit: Could we a test to cover this code path?

File: components/support/utils/src/main/java/mozilla/components/support/utils/ext/WindowInsets.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        @Suppress("DEPRECATION")
+        this.systemWindowInsetRight
+    }
+
+/**
+ * Returns the left system window inset in pixels.
+ */
+fun WindowInsets.left(): Int =
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+        this.getInsets(WindowInsets.Type.systemBars()).left
```
Comment (Amejia481) [inline components/support/utils/src/main/java/mozilla/components/support/utils/ext/WindowInsets.kt:37]: Nit: Could we a test to cover this code path?

File: components/support/utils/src/main/java/mozilla/components/support/utils/ext/WindowInsets.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        @Suppress("DEPRECATION")
+        this.systemWindowInsetTop
+    }
+
+/**
+ * Returns the right system window inset in pixels.
+ */
+fun WindowInsets.right(): Int =
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+        this.getInsets(WindowInsets.Type.systemBars()).right
```
Comment (Amejia481) [inline components/support/utils/src/main/java/mozilla/components/support/utils/ext/WindowInsets.kt:26]: Nit: Could we a test to cover this code path?

File: components/support/utils/src/main/java/mozilla/components/support/utils/ext/WindowInsets.kt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+import android.os.Build
+import android.view.WindowInsets
+
+/**
+ * Returns the top system window inset in pixels.
+ */
+fun WindowInsets.top(): Int =
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+        this.getInsets(WindowInsets.Type.systemBars()).top
```
Comment (Amejia481) [inline components/support/utils/src/main/java/mozilla/components/support/utils/ext/WindowInsets.kt:15]: Nit: Could we a test to cover this code path?

**Review by Amejia481 — APPROVED** [review]
LGTM, lets just add a couple of tests to cover the compact APIs

**Comment by mcarare** [comment]
@Amejia481 I cannot add tests to cover the SDK 30 path because we cannot run tests with SDK 30 until we upgrade robolectric: https://github.com/mozilla-mobile/android-components/issues/9684

**Comment by Amejia481** [comment]
Sorry for the misunderstanding let's land as it is and later add the tests.

**Comment by mcarare** [comment]
No problem! Thanks for the review!

**Pull request merged**